specs/experimental/custom-gas-token.md

```diff
@@ -0,0 +1,471 @@
+
+### StandardBridge
+
+The `StandardBridge` contracts on both L1 and L2 MUST be aware of the custom gas token. The `ether` specific ABI on the
+`StandardBridge` is disabled when custom gas token is enabled.
+
+The following methods MUST revert when custom gas token is being used:
+
+- `bridgeETH(uint32,bytes)`
+- `bridgeETHTo(address,uint32,bytes)`
+- `receive()`
+- `finalizeBridgeETH(address, address, uint256, bytes)`
+
+The following legacy methods MUST also revert when custom gas token is being used:
+
+- `depositETH(uint32,bytes)`
+- `depositETHTo(address,uint32,bytes)`
+- `finalizeETHWithdrawal(address,address,uint256,bytes)`
+- `withdraw(address,uint256,uint32,bytes)`
+- `withdrawTo(address,address,uint256,uint32,bytes)`
+- `finalizeDeposit(address,address,address,address,uint256,bytes)`
+
+### CrossDomainMessenger
+
+The `CrossDomainMessenger` contracts on both L1 and L2 MUST not be able to faciliate the transfer of native asset
+on custom gas token chains due to their tight coupling with `ether` and the `OptimismPortal` and `L2ToL1MessagePasser`
+contracts.
+
+It is possible to support this in the future with a redesign of the `CrossDomainMessenger` contract.
+
+The following methods MUST revert when `CALLVALUE` is non zero:
+
+- `sendMessage(address,bytes,uint32)`
+- `relayMessage(uint256,address,address,uint256,uint256,bytes)`
+
+It MAY be safe to not always directly revert in these implementations if the revert happens elsewhere in the callstack.
+
+### SystemConfig
+
+The `SystemConfig` is the source of truth for the address of the custom gas token. It does on chain validation,
+stores information about the token and well as passes the information to L2.
+
+#### initialize
+
+The `SystemConfig` is modified to allow the address of the custom gas paying token to be set during the call
+to `initialize`. Using a custom gas token is indicated by passing an address other than `ETHER_TOKEN_ADDRESS`
+or `address(0)`. If `address(0)` is used for initialization, it MUST be mapped into `ETHER_TOKEN_ADDRESS`
+before being forwarded to the rest of the system. When a custom gas token is set, the number of decimals
+on the token MUST be exactly 18, the name of the token MUST be less than or equal to 32 bytes and the
+symbol MUST be less than or equal to 32 bytes. If the token passes all of these checks,\
+`OptimismPortal.setGasPayingToken` is called. The implementation of `initialize` MUST not allow the chain
+operator to change the address of the custom gas token if it is already set.
+
+#### Getters
+
+Some getters are added to the `SystemConfig`:
+
+```solidity
+function gasPayingToken() public view returns (address, uint8);
+function gasPayingTokenName() public view returns (string);
+function gasPayingTokenSymbol() public view returns (string);
+```
+
+If custom gas token is not used, then `gasPayingToken()` should return `(ETHER_TOKEN_ADDRESS,18)`,
+`gasPayingTokenName` should return `ether` and `gasPayingTokenSymbol` should return `ETH`.
```

Comment:
not blocking nit since most token names start with a capital, you can reject if the code diff here is annoying this late
```suggestion
`gasPayingTokenName` should return `Ether` and `gasPayingTokenSymbol` should return `ETH`.
```

Reply:
Though one thought here is if we go `ether` then the L2 weth9 predeploy would change to return `Wrapped ether` instead of the current `Wrapped Ether`.

Reply:
Ah you are right cc @0xfuturistic can you double check this is right in the code?

Reply:
should we update the specs to match the code (i.e., stick to `Ether`) or update the code to match the specs (i.e., stick to `ether`). imo, the former feels more consistent and would require not adding additional logic to the code (such as a system that capitalized the name when computing `Wrapped + name`)

Reply:
It should be updated to be `"Ether"` so its consistent with weth9 behavior